internal/gateway/proxy/proxy.go

```diff
@@ -227,6 +231,55 @@ func (cp *ControllerProxy) GetEvent(ctx context.Context,
 	}, nil
 }
 
+func (cp *ControllerProxy) ValidateSubscription(ctx context.Context,
+	req *proxypb.ValidateSubscriptionRequest) (*proxypb.ValidateSubscriptionResponse, error) {
+	if req.GetEvent() == nil {
+		res, err := cp.GetEvent(ctx, &proxypb.GetEventRequest{
+			Eventbus:   req.Eventbus,
+			EventlogId: req.Eventlog,
+			Offset:     req.Offset,
+			Number:     1,
+		})
+		if err != nil {
+			return nil, err
+		}
+		req.Event = res.GetEvents()[0].Value
+	}
+
+	e := v2.NewEvent()
+	if err := e.UnmarshalJSON(req.GetEvent()); err != nil {
+		return nil, errors.ErrInvalidRequest.WithMessage("failed to unmarshall event to CloudEvent").Wrap(err)
+	}
+
+	if req.GetSubscription() == nil {
+		sub, err := cp.GetSubscription(ctx, &ctrlpb.GetSubscriptionRequest{Id: req.SubscriptionId})
+		if err != nil {
+			return nil, err
+		}
+		req.Subscription = &ctrlpb.SubscriptionRequest{
+			Filters:     sub.Filters,
+			Transformer: sub.Transformer,
+		}
+	}
+
+	sub := convert.FromPbSubscriptionRequest(req.Subscription)
+	res := &proxypb.ValidateSubscriptionResponse{}
+	f := filter.GetFilter(sub.Filters)
+	r := f.Filter(e)
+	if !r {
+		return res, nil
+	}
+
+	res.Matched = true
+	t := transform.NewTransformer(sub.Transformer)
+	if err := t.Execute(&e); err != nil {
```

Comment:
need support only has Filters no Transformer

Reply:
The transformer also needs support.

Reply:
I mean if the subscription only has filters and no has transformer,  `transform.NewTransformer(sub.Transformer)`  will return nil, then `t.Execute(&e)` will panic